src/theoretical.jl

```diff
@@ -64,6 +64,27 @@ Evaluate the variogram at points `x` and `y`.
 #------------------
 # IMPLEMENTATIONS
 #------------------
+"""
+    NuggetEfect(n)
+
+A pure nugget effect (random) variogram with nugget `n`
+"""
+@with_kw struct NuggetEffect{T,D} <: Variogram{T,D}
+  nugget::T = 0.0
+  sill::T = 0.0
+  range::T = 0.0
```

Comment:
We don't need to store these fields. We just need the `nugget`. For the sill and range we specialize the methods `sill(γ::Nugget{T,D}) = γ.nugget`.

Reply:
We need them until fitting is fixed. I think is not good to merge code that do not pass tests.

Reply:
No. We need to fix fitting in a separate PR. This PR is about adding a new NuggetEffect struct with the correct definition. We don't want to mix things. Please update the PR following the suggestions.

Reply:
Interesting decision. It is not hard to do what you want, but the test will fail. I will do it.

Reply:
The test shouldn't fail. Please do not add the nugget effect to the list of stationary variogram tests.

Reply:
Could you explain why I should not include it. Thanks.

Reply:
Because these tests were designed for actual variograms models that aren't pure nugget. Unless you plan to improve the fitting methods for additive models and pure nugget _here_, this PR shouldn't forcefully make tests fail. Makes sense?

Reply:
Now with the explanation it makes sense. With the NuggetEffect implemented we can define properly what a nested variogram model is, which is fundamental for TB and later with proper multivariate estimation/simulation implementations.